index.js

```diff
@@ -237,6 +237,37 @@ const execa = (command, args, options) => {
 		return Promise.reject(error);
 	}
 
+	const originalKill = spawned.kill.bind(spawned);
+	spawned.kill = (signal, options) => {
+		if (signal === undefined ||
+			(typeof signal === 'string' && signal.toUpperCase() === 'SIGTERM') ||
+			(typeof signal === 'number' && signal === os.constants.signals.SIGTERM)) {
+			const retryAfter = options !== undefined &&
+				['number', 'string'].includes(typeof options.retryAfter) ?
+				Number(options.retryAfter) :
+				5000;
+
+			const retry = !(options !== undefined && options.retry === false);
+
+			const killResult = originalKill(signal);
+
+			if (retry && killResult) {
+				const safetyTimeout = setTimeout(() => {
+					try {
+						originalKill('SIGKILL');
+					} catch (_) {
+						// All good - process killed
```

Comment:
This comment should be removed. If an error is thrown there, it's most likely that the process is not killed actually. `process.kill()` does not throw when process is already killed or does not exist.

I checked the Node.js source and it's actually quite unlikely the code will reach this line. You should try to test cover it to see if that's even possible. However it's better to be safe than sorry, consider it would otherwise throw an uncaught exception, which could call `process.exit()`.

Reply:
Yes, I mixed this one up with process.kill even though we talk child_process.kill here. My bad. I removed the comment: b8e4bce76a7565f1bfac4c717ccd7fc2a5739108